### ISSUE TITLE
peer = nil issue

### ISSUE THREAD

**Issue by martinlindhe** [macroquest/mq-definitions#27]
The following declaration makes `peer` be a global variable, which is confusing.

https://github.com/macroquest/mq-definitions/blob/1864fe4835c6afd040703576ae33d59cc686cfda/plugins/DanNet/_datatypes.lua#L5

**Comment by martinlindhe** [comment]
![Capture](https://user-images.githubusercontent.com/181531/194604282-0d520edb-f728-4f96-8ede-2342d88c92c5.JPG)

**Comment by projecteon** [comment]
This goes for all the datatypes which has this kind of definition setup to give better documentation and be more clear in the definitions file. Not sure there is a good way around it.

**Comment by brainiac** [comment]
Having same issue with things like `integer` and `float`

These type names need better qualification (e.g. MQinteger and MQfloat)

**Comment by projecteon** [comment]
@brainiac 
Do you have examples of where this is wrong?
`integer`is a part of the sumneko base definitions. 
Im tried to catch all MQInt and MQFloat when I wrote this base type wrappers and having a hard time finding which ones I missed.

I see we do have a float class which probably should be removed as we try to encourage using LUA base types as much as possible.

**Comment by brainiac** [comment]
you're right about integer. I find another example but fixed it, assumed that it was more prevalent.
